platform/engineering/design-docs/2021-08-01-autonomous-deployments.md

```diff
@@ -0,0 +1,211 @@
+
+#### Technologies Being Used
+
+- [Github Actions](https://github.com/features/actions)
+- [AWS](https://docs.aws.amazon.com/AmazonS3/latest/userguide/Welcome.html)
+- [Webpack](https://webpack.js.org/)
+
+### Work Estimates
+
+#### Milestone: Update current system to function with new workflows
+
+- Update current (dev/staging) deploy scripts to sync and not delete files from s3 OR create new script
+- Create single app build in webpack that includes assets.
+
+This work should be completed in 1 sprint with 2 people (1 person on each task).
+
+#### Milestone: Create and release staging/dev deployment behaviors
+
+- Create deployment script for dev
+- Disable current dev deployment behaviors
+- Announce and reserve time for troubleshooting dev before moving to staging
+- Create deployment script for staging
+- Disable current staging deployment behaviors
+- Reserve time for troubleshooting staging
+- Reach out to two application teams for production deployment beta
+
+This work should be completed in 2-3 sprints with 2 people. Sprint 1 should be dev. Sprint 2 should be staging. Reserving and extra sprint for troubleshooting if needed.
+
+#### Milestone: Create and release initial production behaviors with "allow" list
+
+- Create "allow" list for applications
+- Create production deployment workflow
+- Monitor auto and manual deployments for a week to debug and troubleshoot
+
+This work should be completed in 2-3 sprints with 2 people, one of which should be very familiar with Github Actions.
+
+#### Milestone: Create and release MVP production behaviors with "disallow" list
+
+- Update deployment workflow to use disallow list instead of allow list
+- Plan and execute VFS training on the update
+- Release to teams
+- Monitor auto and manual deployments for a sprint to debug and troubleshoot
+
+This work should be completed in 2-3 sprints with 2 people.
+
+### Alternatives
+
+An alternative that was discussed was isolating the applications in their own repositories before executing autonomous deploys. We felt that this would bloat the timeline and delay the feature from being released for too long.
+
+We also discussed monorepo isolation, but deemed it as unnecessary to complete the objective.
+
+### Post MVP
+
+Initial dashboard view where teams sign in with Github auth, view their apps and manage their deployments including: scheduling, reverting, and manual deployment. This dashboard and infrastructure should be scoped and estimated separately as a post-MVP item. It will be the basis for a future platform dashboard.
+
+The goal of this post-MVP stage is to abstract the Github UI from the deployment actions so users are unaware of what is under the hood.
+
+After autonomous deployment work has been completed, work on application isolation will begin and another design document will be created.
+
+We would also like to be able to track which version of an app is currently released.
```

Comment:
@k80bowman added this as a post-mvp item. I think it's a good point to have. If we come up with a solution we can try and add it into the main doc

Reply:
Thanks for adding this. Yes, it is going to get very weird.